diff --git a/senn/package/fcitx-senn/CMakeLists.txt b/senn/package/fcitx-senn/CMakeLists.txt
index e0678c2..070b4bb 100644
--- a/senn/package/fcitx-senn/CMakeLists.txt
+++ b/senn/package/fcitx-senn/CMakeLists.txt
@@ -21,7 +21,7 @@ include_directories(
   ${FCITX4_FCITX_UTILS_INCLUDE_DIRS}
   ../../third-party/
   ${SENN_CPP_DIR}
-#  $ENV{HOME}/.roswell/impls/x86-64/linux/ecl/21.2.1/include
+  $ENV{HOME}/.roswell/impls/x86-64/linux/ecl/21.2.1/include
   )
 
 # $ mkdir dep
@@ -38,22 +38,22 @@ fcitx_add_addon_full(senn
           ${SENN_CPP_DIR}/ipc/ipc.cpp
           ${SENN_CPP_DIR}/process/process.cpp
           ${SENN_CPP_DIR}/senn_fcitx/im/stateful_ime_proxy.cpp
-          ${SENN_CPP_DIR}/senn_fcitx/im/stateful_ime_ipc.cpp
-#          ${SENN_CPP_DIR}/senn_fcitx/im/stateful_ime_ecl.cpp
-#  LINK_LIBS ${CMAKE_CURRENT_SOURCE_DIR}/dep/senn-lib-fcitx--all-systems.a
-#            ${CMAKE_CURRENT_SOURCE_DIR}/dep/libecl.so
+#          ${SENN_CPP_DIR}/senn_fcitx/im/stateful_ime_ipc.cpp
+          ${SENN_CPP_DIR}/senn_fcitx/im/stateful_ime_ecl.cpp
+  LINK_LIBS ${CMAKE_CURRENT_SOURCE_DIR}/dep/senn-lib-fcitx--all-systems.a
+            ${CMAKE_CURRENT_SOURCE_DIR}/dep/libecl.so
   IM_CONFIG senn.conf)
 
 
-# install(
-#   PROGRAMS ${CMAKE_CURRENT_SOURCE_DIR}/dep/kkc-engine
-#   CONFIGURATIONS Release
-#   RUNTIME DESTINATION "/usr/lib/senn")
+install(
+  PROGRAMS ${CMAKE_CURRENT_SOURCE_DIR}/dep/kkc-engine
+  CONFIGURATIONS Release
+  RUNTIME DESTINATION "/usr/lib/senn")
 
-# install(
-#   PROGRAMS ${CMAKE_CURRENT_SOURCE_DIR}/dep/libecl.so.21.2 ## Seems that CMake add this specific file of dependency
-#   CONFIGURATIONS Release
-#   RUNTIME DESTINATION "/usr/lib/senn")
+install(
+  PROGRAMS ${CMAKE_CURRENT_SOURCE_DIR}/dep/libecl.so.21.2 ## Seems that CMake add this specific file of dependency
+  CONFIGURATIONS Release
+  RUNTIME DESTINATION "/usr/lib/senn")
 
 install(
   PROGRAMS ${SENN_CPP_DIR}/gui/menu/about/about
diff --git a/senn/package/fcitx-senn/fcitx_senn.cpp b/senn/package/fcitx-senn/fcitx_senn.cpp
index 62ba068..bab4e0b 100644
--- a/senn/package/fcitx-senn/fcitx_senn.cpp
+++ b/senn/package/fcitx-senn/fcitx_senn.cpp
@@ -8,8 +8,8 @@
 #include <fcitx/instance.h>
 
 #include "process/process.h"
-#include "senn_fcitx/im/stateful_ime_ipc.h"
-// #include "senn_fcitx/im/stateful_ime_ecl.h"
+// #include "senn_fcitx/im/stateful_ime_ipc.h"
+#include "senn_fcitx/im/stateful_ime_ecl.h"
 // #include "senn_fcitx/im/stateful_ime_sbcl.h"
 
 namespace {
@@ -276,9 +276,7 @@ static void FcitxSennDestroy(void *arg) {
   FcitxSenn *senn = (FcitxSenn *)arg;
 
   delete senn->ime;
-  /*
   senn::fcitx::im::StatefulIMEEcl::ClShutdown();
-  */
 
   senn::fcitx_senn::menu::Destory(senn->fcitx, &senn->menu);
 
@@ -295,17 +293,17 @@ static void *FcitxSennCreate(FcitxInstance *fcitx) {
   senn->fcitx = fcitx;
 
   // StatefulIME
+  /*
   senn->ime =
       senn::fcitx::im::StatefulIMEIPC::SpawnAndCreate("/usr/lib/senn/server");
+  */
   /*
   senn::fcitx::im::StatefulIMESbcl::Init("/usr/lib/senn/libsennfcitx.core");
   senn->ime = senn::fcitx::im::StatefulIMESbcl::Create()
   */
-  /*
   senn::fcitx::im::StatefulIMEEcl::ClBoot();
   senn::fcitx::im::StatefulIMEEcl::EclInitModule();
   senn->ime = senn::fcitx::im::StatefulIMEEcl::Create();
-  */
 
   FcitxIMEventHook hk;
   hk.arg = senn;
